### PR TITLE
Scala 2.13.0

### PR DESCRIPTION
**Pull request by balagez**
Hey! Can you pease review, merge and publish?

### CODE REVIEW

**Comment by Ma27** [comment]
After skimming through the diff the change seems fine. You don't have to be confused if travis-ci fails, the test suite is somehow broken atm (it's about to get fixed in #17).

I'll do some more extensive testing (haven't even read the Scala 2.13 changelog tbh) during the next week, but I'm pretty sure that we can merge this without further effort then.

ideally we'd get this **and** our test-suite stabilization ready before I publish a new release, as I'm pretty unhappy about the current situation of your test-suite. But let's see, if that takes too much time, I'd also create a release before #17 being ready to get this shipped :+1:

File: src/main/scala/redis/api/Clusters.scala [hunk 1]
```diff
@@ -44,7 +44,7 @@ case class ClusterSlots() extends RedisCommand[MultiBulk,Seq[ClusterSlot]] {
           case MultiBulk(Some(responses)) =>
               responses.grouped(3).map { vect =>
                 buildClusterNode(vect)
-              }.toSeq
+              }.toList
```
Comment (balagez) [inline src/main/scala/redis/api/Clusters.scala:47]: `.toSeq` here returned `Stream` prior to 2.13 so I changed it to `.toList` to be compatible with all Scala versions at the same time. See the change in `RedisClusterTest.scala`.

File: src/main/scala/redis/commands/Transactions.scala [hunk 2]
```diff
@@ -112,12 +112,10 @@ case class Transaction(watcher: Set[String], operations: Queue[Operation[_, _]],
 
   def dispatchExecReply(multiBulk: MultiBulk) = {
     multiBulk.responses.map(replies => {
-      (replies, operations).zipped.map((reply, operation) => {
-        reply match {
-          case e: Error => operation.completeFailed(ReplyErrorException(e.toString()))
-          case _ => operation.tryCompleteSuccess(reply)
-        }
-      })
+      replies.zip(operations).map {
```
Comment (balagez) [inline src/main/scala/redis/commands/Transactions.scala:115]: The 2.13 compiler recommends using `.lazyZip` here but that's not backward compatible and I think we don't need lazy evaluation here.

File: src/test/scala/redis/RedisSpec.scala [hunk 3]
```diff
@@ -12,7 +12,7 @@ import org.specs2.concurrent.FutureAwait
 import org.specs2.mutable.SpecificationLike
 import org.specs2.specification.core.Fragments
 
-import scala.collection.JavaConverters._
+// import scala.collection.JavaConverters._
```
Comment (balagez) [inline src/test/scala/redis/RedisSpec.scala:13]: This is renamed in Scala 2.13 and the new import is not backward compatible but fortunately the code that used it was also commented out.

File: src/main/scala/redis/Sentinel.scala [hunk 4]
```diff
@@ -190,7 +190,7 @@ abstract class SentinelMonitored(system: ActorSystem, redisDispatcher: RedisDisp
   def withSlavesAddr[T](initFunction: Seq[(String, Int)] => T): T = {
     import scala.concurrent.duration._
 
-    val fslaves = Future.sequence(sentinelClients.values.map(_.slaves(master)))
+    val fslaves = Future.sequence(sentinelClients.map({ case (_, client) => client.slaves(master) }))
```
Comment (balagez) [inline src/main/scala/redis/Sentinel.scala:193]: Using `.values` here broke `SentinelMonitoredRedisClientMasterSlavesSpec` with Scala 2.13. I didn't look into the exact reason but using `.values` was discouraged even before Scala 2.13 so just to be on the safe side I replaced all of them with `.map`.

**Review by balagez — COMMENTED** [review]
Added a few comments to explain some of the changes.

**Review by herzrasen — APPROVED** [review]
LGTM. Thanks for your effort @balagez

**Comment by KarelCemus** [comment]
Is there any plan to merge this and release a new version? It would be great.

**Comment by ghost** [comment]
> Is there any plan to merge this and release a new version? It would be great.

Seconded.

**Comment by Ma27** [comment]
FYI it's definetely planned to release this during one of the next two weeks, I'll hopefully be able to reserve sufficient time to actually test this before :)

**Comment by endercrest** [comment]
> FYI it's definetely planned to release this during one of the next two weeks, I'll hopefully be able to reserve sufficient time to actually test this before :)

Any update on this?

**Comment by Ma27** [comment]
Not yet :/ I'll do my best to publish a release this week :)

**Comment by KarelCemus** [comment]
Any progress on this?

**Comment by nehaev** [comment]
@Ma27 Any updates?

**Comment by kardapoltsev** [comment]
Changes are look good to me but lot of them aren't related to scala 2.13 update. I think it's better to make another commits for them.

**Comment by balagez** [comment]
@kardapoltsev They are all related to Scala 2.13, I did the absolute minimum to make this project compile with 2.13 and the tests not fail.

Edit: But true, many of them can be rolled out before actually adding 2.13 to build.sbt but I doubt that having to wait for two PRs to be tested and merged separately would speed up anything.

**Comment by Ma27** [comment]
I'll rebase the conflicts locally and see if everything remains working.

**Comment by kardapoltsev** [comment]
Glad to see you back @Ma27 (:

**Comment by kardapoltsev** [comment]
> @kardapoltsev They are all related to Scala 2.13, I did the absolute minimum to make this project compile with 2.13 and the tests not fail.
> 
> Edit: But true, many of them can be rolled out before actually adding 2.13 to build.sbt but I doubt that having to wait for two PRs to be tested and merged separately would speed up anything.

Yes, my bad, sorry for that. I wasn't thoughtful when going thought your changes.

**Comment by Ma27** [comment]
Fixed most of the conflicts locally (stil having some test-suite issues though)... Will catch some sleep now and fix the rest tomorrow... as this is the only thing that should be done before a new release, I'd also prepare the release tomorrow in case of suficient time :)

**Comment by kardapoltsev** [comment]
@Ma27 any updates? May I help you with this PR?

**Comment by Ma27** [comment]
Not sure.... had a hard time resolving conflicts for this patch (and was busy with a lot of other stuff :/). If my local changes are sufficiently functional, I'd push the rebased version tomorrow and publish a new release as soon as I have time to do this.

**Comment by kardapoltsev** [comment]
ping @Ma27

**Comment by Ma27** [comment]
Tests seem fine now. Let's wait for travis and merge. After that I'll prepare a release.

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
Note: temporary disabled 2.13 in the build as this causes some dependency conflicts. I'm looking for an appropriate solution and publish a release after this for 2.{11,12,13}.

**Comment by Ma27** [comment]
Very sorry for the long delay guys... just release 1.9.0 and started the release process at maven central :)

**Pull request merged**